mbed/mbed.py

```diff
@@ -1223,7 +1225,9 @@ def clone(self, url, path, rev=None, depth=None, protocol=None, offline=False, *
                         os.makedirs(os.path.split(path)[0])
 
                     info("Carbon copy from \"%s\" to \"%s\"" % (cache, path))
+                    self.cache_lock(url)
                     shutil.copytree(cache, path)
+                    self.cache_unlock(url)
```

Comment:
Could this be safer as a context manager? That way we could never forget to unlock the cache.